docs/root/intro/arch_overview/http/http_proxy.rst

```diff
@@ -0,0 +1,38 @@
+.. _arch_overview_http_dynamic_forward_proxy:
+
+HTTP dynamic forward proxy
+==========================
+
+.. attention::
+
+  HTTP dynamic forward proxy support should be considered alpha and not production ready. Stats
+  as well as circuit breakers are missing and will be added soon.
+
+Though the combination of both an :ref:`HTTP filter <config_http_filters_dynamic_forward_proxy>` and
+:ref:`custom cluster <envoy_api_msg_config.cluster.dynamic_forward_proxy.v2alpha.ClusterConfig>`,
+Envoy supports HTTP dynamic forward proxy. This means that Envoy can perform the role of an HTTP
+proxy without prior knowledge of all configured DNS addresses, while still retaining the vast
+majority of Envoy's benefits including asynchronous DNS resolution. The implementation works as
+follows:
+
+* The dynamic forward proxy HTTP filter is used to pause requests if the target DNS host is not
+  already in cache.
+* Envoy will begin asynchronously resolving the DNS address, unblocking any requests waiting on
+  the response.
+* Any future requests will not be blocked as the DNS address is already in cache. The resolution
+  process works similarly to the :ref:`logical DNS
+  <arch_overview_service_discovery_types_logical_dns>` service discovery type with a single target
```

Comment:
curious. why not strict dns?

Reply:
See the discovery type docs, https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#arch-overview-service-discovery-types-logical-dns logical_dns:

> instead of strictly taking the results of the DNS query and assuming that they comprise the entire upstream cluster, a logical DNS cluster only uses the first IP address returned when a new connection needs to be initiated. Thus, a single logical connection pool may contain physical connections to a variety of different upstream hosts. Connections are never drained. This service discovery type is optimal for large scale web services that must be accessed via DNS. Such services typically use round robin DNS to return many different IP addresses. Typically a different result is returned for each query.